### ISSUE TITLE
$auth.currentUser doesn't update when subscribe to new data

### ISSUE THREAD

**Issue by jamalx31** [Urigo/angular-meteor#1267]
In my application at some page I subscribe to new date like this: (angular 1.3.5)
```
Meteor.publish("userData", function () {
  if (this.userId) {
    return Meteor.users.find({_id: this.userId},
                             {fields: {'other': 1, 'things': 1}});
  } else {
    this.ready();
  }
});
```

but $auth.currentUser doesn't get it while $root.currentUser does

**Comment by ThomsCass** [comment]
Can you be more explicit on the workflow ?

**Comment by jamalx31** [comment]
@ThomsCass 
The default fields the server publish for currentUser are "id" and "profile", on a page in my application
I call for another subscription to publish another date about currentUser like I said before.
On that page where I call the new subscription, "$root.currentUser" gets the new fields, but "$auth.currentUser" doesn't

**Comment by Urigo** [comment]
@jamalx31 it's a big hard to get to the bottom of it without a reproduction.

Can you create a small reproduction of that problem in a separate repo and link to it?

Thank you

**Comment by jamalx31** [comment]
Hi @Urigo 
https://github.com/jamalx31/meteor-rep
in this repo you can reproduce the bug, just run it with ```meteor``` and you will see it

**Comment by kamilkisiela** [comment]
@jamalx31 You can update your app to use `angular-meteor-auth@1.0.2_1` and replacing `$auth.currentUser` with `vm.currentUser` or `currentUser` if you're using it with $scope.

**Comment by jamalx31** [comment]
@kamilkisiela @Urigo 
it seems like $rootScope.currentUser and $scope.currentUser are not the same object
in $scope.currentUser I miss some of the fields I publish for current user!

**Comment by Urigo** [comment]
released `1.0.3`.  please re-open if it's not fixed for you
